packages/grid/src/Grid.tsx

```diff
@@ -1314,26 +1323,52 @@ class Grid extends PureComponent<GridProps, GridState> {
       }
     }
 
-    this.setViewState({ top, left, topOffset, leftOffset });
+    this.setViewState({ top, left, topOffset, leftOffset }, false, event);
   }
 
   /**
    * Checks the `top` and `left` properties that are set and updates the isStuckToBottom/Right properties
    * Should be called when user interaction occurs
    * @param viewState New state properties to set.
    * @param forceUpdate Whether to force an update.
+   * @param event The event that triggered the update.
    */
-  setViewState(viewState: Partial<GridState>, forceUpdate = false): void {
+
+  setViewState(
+    viewState: Partial<GridState>,
+    forceUpdate = false,
+    event?: WheelEvent | GridKeyboardEvent
+  ): void {
     if (!this.metrics) throw new Error('metrics not set');
 
     const { isStickyBottom, isStickyRight } = this.props;
     const { top, left } = viewState;
     const { lastTop, lastLeft } = this.metrics;
+    let isUserInputDown = false;
+    let isUserInputRight = false;
+
+    if (event instanceof WheelEvent) {
+      isUserInputDown = event.deltaY > 0;
+      isUserInputRight = event.deltaX > 0;
+    } else if (
+      event instanceof KeyboardEvent ||
+      (event && 'nativeEvent' in event) // used to catch the case that a synthetic react keyboard event is passed
+    ) {
+      isUserInputDown =
+        event.key === 'ArrowDown' ||
+        event.key === 'End' ||
+        event.key === 'PageDown';
+      isUserInputRight = event.key === 'ArrowRight';
+    }
```

Comment:
I know I said to do these checks here in the last review - but this logic doesn't really belong here. `SelectionKeyHandler` has all the logic for which keys map to which direction, and arguably the logic from `handleWheel` for scrolling should be extracted to it's own mouse handler.

The reason I mention this though is because there are cases that aren't handled here - for example, if you press Shift+End, it will go to the far right instead of the bottom.

So instead of taking an event and passing it through, we really should just be passing in an option, e.g.
```
type StickyOptions = { shouldStickRight: boolean; shouldStickBottom: boolean };
setViewState(..., stickyOptions?: StickyOptions)
```